tesp_api/service/event_actions.py

```diff
@@ -156,94 +176,119 @@ async def handle_run_task(event: Event) -> None:
+        command_status: dict
 
-        # start the task (docker container/s) in the pulsar
-        await pulsar_operations.run_job(task_id, run_command)
-
-        # wait for the task
-        command_status = await pulsar_operations.job_status_complete(str(task_id))
+        if run_command_str is None:
+            print(f"Task {task_id} has no commands to run. Treating as successful no-op.")
+            command_status = {'stdout': '', 'stderr': 'No commands to run.', 'returncode': 0}
+        else:
+            print(f"Submitting job to Pulsar for task {task_id}: {run_command_str}")
+            await pulsar_operations.run_job(task_id, run_command_str)
+            command_status = await pulsar_operations.job_status_complete(str(task_id)) # Polls Pulsar for job completion
 
         command_end_time = datetime.datetime.now(datetime.timezone.utc)
         await append_task_executor_logs(
-            task_id,
-            author,
-            TesTaskState.RUNNING,
-            command_start_time,
-            command_end_time,
-            command_status['stdout'],
-            command_status['stderr'],
-            command_status['returncode']
+            task_id, author, TesTaskState.RUNNING,
+            command_start_time, command_end_time,
+            command_status.get('stdout', ''), command_status.get('stderr', ''),
+            command_status.get('returncode', -1)
         )
-        if command_status['returncode'] != 0:
-            task = await task_repository.update_task_state(
-                task_id,
-                TesTaskState.RUNNING,
-                TesTaskState.EXECUTOR_ERROR
-            )
-
-            raise TaskExecutorError()
 
-    except Exception as error:
-        pulsar_event_handle_error(error, task_id, event_name, pulsar_operations)
-
-    #    dispatch_event('finalize_task', payload)
-    await Promise(lambda resolve, reject: resolve(None)) \
-        .then(lambda ignored: task_repository.update_task_state(
-            task_id,
-            TesTaskState.RUNNING,
-            TesTaskState.COMPLETE
-        )) \
-        .map(lambda task: get_else_throw(
-            task, TaskNotFoundError(task_id, Just(TesTaskState.RUNNING))
+        # Re-fetch task state to check for external cancellation during job execution
+        current_task_monad = await task_repository.get_task(maybe_of(author), {'_id': task_id}) 
+        current_task_obj = get_else_throw(current_task_monad, TaskNotFoundError(task_id)) 
+
+        if current_task_obj.state == TesTaskState.CANCELED:
+            print(f"Task {task_id} found CANCELED after job completion polling. Aborting state changes.")
+            return 
+
+        if command_status.get('returncode', -1) != 0:
+            print(f"Task {task_id} executor error (return code: {command_status.get('returncode', -1)}). Setting state to EXECUTOR_ERROR.")
+            await task_repository.update_task_state(task_id, TesTaskState.RUNNING, TesTaskState.EXECUTOR_ERROR)
```

Comment:
Or does it work now logically in the same way as before with the TesTaskState.EXECUTOR_ERROR ?